Source/Core/Core/IOS/Network/NCD/Manage.cpp

```diff
@@ -29,11 +35,51 @@ IPCCommandResult NetNCDManage::IOCtlV(const IOCtlVRequest& request)
   switch (request.request)
   {
   case IOCTLV_NCD_LOCKWIRELESSDRIVER:
+    if (!request.HasNumberOfValidVectors(0, 1))
+      return GetDefaultReply(IPC_EINVAL);
+
+    if (request.io_vectors[0].size < 2 * sizeof(u32))
+      return GetDefaultReply(IPC_EINVAL);
+
+    if (m_context.ipc_fd != 0)
+    {
+      // It is an error to lock the driver again when it is already locked.
+      common_result = IPC_EINVAL;
+    }
+    else
+    {
+      // NCD writes the internal address of the request's file descriptor.
+      // We will just write the value of the file descriptor.
+      // The value will be positive so this will work fine.
+      m_context.ipc_fd = request.fd;
+      Memory::Write_U32(request.fd, request.io_vectors[0].address + 4);
+    }
     break;
 
   case IOCTLV_NCD_UNLOCKWIRELESSDRIVER:
-    // Memory::Read_U32(request.in_vectors.at(0).address);
+  {
+    if (!request.HasNumberOfValidVectors(1, 1))
+      return GetDefaultReply(IPC_EINVAL);
+
+    if (request.in_vectors[0].size < sizeof(u32))
+      return GetDefaultReply(IPC_EINVAL);
+
+    if (request.io_vectors[0].size < sizeof(u32))
```

Comment:
Did you mean to check the same statement twice?

Reply:
No, that's a good catch!

Reply:
They are separate variables.  Sorry, `in_vectors` and `io_vectors` looks very similar...